cirq/ops/two_qubit_diagonal_gate_test.py

```diff
@@ -0,0 +1,115 @@
+# Copyright 2018 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import itertools
+import numpy as np
+import pytest
+import sympy
+
+import cirq
+
+
+@pytest.mark.parametrize('gate,ignoring_global_phase',
+                         ((cirq.TwoQubitDiagonalGate([2, 3, 5, 7]), True),
+                          (cirq.TwoQubitDiagonalGate([0, 0, 0, 0]), True)))
+def test_consistent_protocols(gate, ignoring_global_phase):
+    cirq.testing.assert_implements_consistent_protocols(
+        gate, ignoring_global_phase=ignoring_global_phase)
+
+
+def test_unitary():
+    diagonal_angles = [2, 3, 5, 7]
+    assert cirq.has_unitary(cirq.TwoQubitDiagonalGate(diagonal_angles))
+    np.testing.assert_allclose(
+        cirq.unitary(cirq.TwoQubitDiagonalGate(diagonal_angles)),
+        np.diag([np.exp(1j * angle) for angle in diagonal_angles]),
+        atol=1e-8)
+
+
+@pytest.mark.parametrize('op,max_two_cost', [
+    (cirq.TwoQubitDiagonalGate([2, 3, 5, 7])(cirq.LineQubit(1),
+                                             cirq.LineQubit(2)), 1),
+])
+def test_decomposition_cost(op: cirq.Operation, max_two_cost: int):
+    ops = tuple(
+        cirq.flatten_op_tree(cirq.google.ConvertToXmonGates().convert(op)))
+    two_cost = len([e for e in ops if len(e.qubits) == 2])
+    over_cost = len([e for e in ops if len(e.qubits) > 2])
+    assert over_cost == 0
+    assert two_cost == max_two_cost
+
+
+@pytest.mark.parametrize('gate', [
+    cirq.TwoQubitDiagonalGate([2, 3, 5, 7]),
+])
+def test_decomposition_respects_locality(gate):
```

Comment:
This test is redundant for a two qubit gate.

Reply:
Done